### PR TITLE
feat: add initial logging for backend

### LINKED ISSUE #511: Logging for backend
### Is your feature request related to a problem? Please describe.

we should log what happens in our backend system into a file, change our `print` to appropriate logging functions

### Describe the solution you'd like

basic logging, to trace errors better

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nullxDEADBEEF**
# Description

Please include a summary of the change and which issue is fixed.

Adds initial version of logging for our backend

Fixes #511

### CODE REVIEW

File: backend/cli.py [hunk 1]
```diff
@@ -42,7 +42,8 @@ def add_imdb_ratings(min_votes: int = 50):
     imdb_ratings = {}
 
     if not res.status_code == 200:
-        echo_warning("Failed to download imdb file")
+        log.info("Failed to download IMDb file")
```
Comment (AndreasPB) [inline backend/cli.py:45]: Warning

File: backend/cli.py [hunk 2]
```diff
@@ -232,6 +233,7 @@ def update_media(chunk_size: int, popularity: float = 1):
                     "http://internal:8888/update-media", json={"ids": id_chunk}
                 )
                 if res.status_code != 200:
+                    log.info(res.text)
```
Comment (AndreasPB) [inline backend/cli.py:236]: Warning

File: backend/app/api.py [hunk 3]
```diff
@@ -72,22 +73,26 @@ def fetch_jsongz_files():
...
+                )
         else:
             # if we have tried to get data for more than 30 days we give up
             if day >= 30:
-                print("No downloads for 30 days - giving up")
+                log.info("No downloads for 30 days - giving up")
                 exit(1)
 
-            print("Downloads failed - trying 1 day earlier")
+            log.info("Downloads failed - trying 1 day earlier")
```
Comment (AndreasPB) [inline backend/app/api.py:95]: Warning maybe

File: backend/app/api.py [hunk 4]
```diff
@@ -116,7 +121,7 @@ def fetch_media_ids(
             elif "person" in file:
                 person_ids = list(map(lambda x: f"p{x['id']}", filtered))
             else:
-                print('Filename doesn\'t start with "movie", "tv" or "person"')
+                log.info('Filename doesn\'t start with "movie", "tv" or "person"')
```
[the remark this answers is not in the document]
Comment (nullxDEADBEEF) [reply backend/app/api.py]: Done

File: backend/cli.py (hunk 1 above)
Comment (nullxDEADBEEF) [reply backend/cli.py]: Done

File: backend/cli.py (hunk 2 above)
Comment (nullxDEADBEEF) [reply backend/cli.py]: Done

File: backend/app/api.py (hunk 3 above)
Comment (nullxDEADBEEF) [reply backend/app/api.py]: Set as warning

**Review by AndreasPB — APPROVED** [review]

**Pull request merged**